### ISSUE TITLE
Interpolate Track values on curve

### ISSUE THREAD

**Issue by akre54** [theatre-js/theatre#419]
Hi there - I'd like to get all values for a given Track to be able to add a "trail" effect without statefully adding to a list of "seen" values.

Currently the `types` support a linear interpolator that maps between a beginning and end value, but the result doesn't align with the bezier curve of the timeline's actual keyframes. I'd like to be able to be able to [retrieve a value for a given time](https://github.com/theatre-js/theatre/issues/330#issuecomment-1411080554) or a quantized list of values for the whole sequence time, similar to [D3's interpolateBasis](https://observablehq.com/@d3/d3-interpolatebasis). Currently it requires a fair bit of hacking in userland. Is this something that could be added to the library?

**Comment by AriaMinaei** [comment]
This sounds like a great feature to have. Maybe `seq.getKeyframes(object.props.x) => Array<Keyframe>` or something like that. It's not yet on the roadmap but I'll be happy to merge a PR if one shows up ;)

**Comment by akre54** [comment]
Yes I can definitely look into this - finding it a bit tricky to wade through that part of the code but I'll try to whip something up.

**Comment by akre54** [comment]
Created: #426
